I suppose that unknown-attributes warning could be disabled, I am continuing on with this warning disabled in libthr

Ah OK, so I also hit the same problem in rtld-elf too

```
--- rtld.o ---
rtld.c:333:17: error: unknown attribute '__nonstring__' ignored [-Werror,-Wunknown-attributes]
  333 |         __attribute__((__nonstring__)) static const char rtld_utrace_sig[RTLD_UTRACE_SIG_SZ] = RTLD_UTRACE_SIG;
      |                        ^~~~~~~~~~~~~
```

I will also try disabling this warning and continuing on, see how far I get

--- a/pkgs/os-specific/bsd/freebsd/pkgs/libthr.nix
+++ b/pkgs/os-specific/bsd/freebsd/pkgs/libthr.nix
@@ -1,6 +1,7 @@
 {
   lib,
   stdenv,
+  fetchpatch,
   mkDerivation,
   libcMinimal,
   include,
@@ -31,6 +32,19 @@ mkDerivation {
     include
     libgcc
   ];
+
+  patches = [
+    # https://github.com/freebsd/freebsd-src/pull/1882
+    (fetchpatch {
+      name = "freebsd-libthr-use-nonstring-attribute.patch";
+      url = "https://github.com/freebsd/freebsd-src/pull/1882/commits/650800993deb513dc31e99ef5cdecd50ee70bb04.diff";
+      hash = "sha256-WKN7dfGAs1+XADT4aLUkkKmQQ4n7gsyFUTCeo6mcuMY=";
+      includes = [ "lib/libthr/thread/thr_printf.c" ];
+    })
+  ];
+
+  # Presumably newer Clang has gotten more strict.
+  CWARNEXTRA = "-Wno-cast-function-type-mismatch";
 
   preBuild = ''
     export NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE -B${csu}/lib"

--- a/pkgs/os-specific/bsd/freebsd/pkgs/rtld-elf.nix
+++ b/pkgs/os-specific/bsd/freebsd/pkgs/rtld-elf.nix
@@ -1,5 +1,6 @@
 {
   mkDerivation,
+  fetchpatch,
   include,
   rpcgen,
   flex,
@@ -29,6 +30,16 @@ mkDerivation {
     "sys/crypto"
   ]
   ++ extraSrc;
+
+  patches = [
+    # https://github.com/freebsd/freebsd-src/pull/1882
+    (fetchpatch {
+      name = "freebsd-rtld-use-nonstring-attribute.patch";
+      url = "https://github.com/freebsd/freebsd-src/pull/1882/commits/650800993deb513dc31e99ef5cdecd50ee70bb04.diff";
+      hash = "sha256-V9jDE/5Fu6hLIzlG1e6AqLnGwlzW2OjonyUgvSVtm58=";
+      includes = [ "libexec/rtld-elf/rtld.c" ];
+    })
+  ];
 
   outputs = [
     "out"

